src/sagemaker/model_card/model_card.py

```diff
@@ -208,6 +211,257 @@ def search_model_associated_model_cards(model_id: str):
+
+    def __init__(self, user_profile_name: Optional[str] = None):
+        """Initialize the ModelPackageCreator object.
+
+        Args:
+            user_profile_name (str, optional): The name of the user's profile.
+        """ # noqa E501 # pylint: disable=line-too-long
+        self.user_profile_name = user_profile_name
+
+
+class SourceAlgorithm(_DefaultToRequestDict, _DefaultFromDict):
+    """Source algorithm object"""
+
+    def __init__(self, algorithm_name: str, model_data_url: Optional[str] = None):
+        """Initialize source algorithm object
+
+        Args:
+            algorithm_name (str): The ARN of an algorithm resource that was used to create the model package.
+            model_data_url (str, optional): The Amazon S3 path where the model artifacts, which result from model training, are stored.
+        """
+        self.algorithm_name = algorithm_name
+        self.model_data_url = model_data_url
+        
+
+class Container(_DefaultToRequestDict, _DefaultFromDict):
+    """Inference container to store the information like model data urls, images and nearest model names"""
+    def __init__(self, image: str, model_data_url: Optional[str] = None, nearest_model_name: Optional[str] = None):
+        """Initialize inference container object.
+        
+        Args:
+            image (str): The Amazon EC2 Container Registry (Amazon ECR) path where inference code is stored. Also known as inference environment.
+            model_data_url (str, optional): The Amazon S3 path where the model artifacts, which result from model training, are stored. Also known as model artifact
+            nearest_model_name (str, optional): The name of a pre-trained machine learning benchmarked by Amazon SageMaker Inference Recommender model that matches your model.
+        """
+        self.image = image
+        self.model_data_url = model_data_url
+        self.nearest_model_name = nearest_model_name
+
+
+class InferenceSpecification(_DefaultToRequestDict, _DefaultFromDict):
+    """Details about inference jobs that can be run with models based on this model package."""
+    containers = _IsList(Container, ENVIRONMENT_CONTAINER_IMAGES_MAX_SIZE)
+
+    def __init__(self, containers: List[Container]):
+        """Initialize Inference specification object.
+
+        Args:
+            containers (list[Containers]): The list of inference containers.
+        """
+        self.containers = containers
+
+
+class ModelPackage(_DefaultToRequestDict, _DefaultFromDict):
+    """
+    Model package version details
+    """
+
+    model_package_status = _OneOf(ModelPackageStatusEnum)
+    model_approval_status = _OneOf(ModelApprovalStatusEnum)
+    created_by = _IsModelCardObject(ModelPackageCreator)
+    source_algorithms = _IsList(SourceAlgorithm, SOURCE_ALGORITHMS_MAX_SIZE)
+    inference_specification = _IsModelCardObject(InferenceSpecification)
+
+    model_package_optional_fields_mapping = {
```

Comment:
let's use upper case `model_package_optional_fields_mapping` since it is a class level variable. (be consistent with the [existing code](https://github.com/shrestha-bikash/sagemaker-python-sdk/blob/master/src/sagemaker/model_card/model_card.py#L829)